cdap-ui/cypress/integration/pluginjsoncreator.spec.ts

```diff
@@ -0,0 +1,651 @@
+        });
+
+        // Validate whether input has been filled
+        cy.get(`${filterSelector} ${dataCy('filter-name')}`)
+          .invoke('val')
+          .then((val) => {
+            expect(val).equals(filter.name);
+          });
+        if (filter.condition.expression) {
+          cy.get(`${filterSelector} ${dataCy('filter-condition-input')} ${dataCy('expression')}`)
+            .invoke('val')
+            .then((val) => {
+              expect(val).equals(filter.condition.expression);
+            });
+        } else {
+          cy.get(`${filterSelector} ${dataCy('filter-condition-input')} ${dataCy('property')}`)
+            .invoke('val')
+            .then((val) => {
+              expect(val).equals(filter.condition.property);
+            });
+          cy.get(`${filterSelector} ${dataCy('filter-condition-input')} ${dataCy('operator')}`)
+            .invoke('val')
+            .then((val) => {
+              expect(val).equals(filter.condition.operator);
+            });
+          if (filter.value) {
+            cy.get(`${filterSelector} ${dataCy('filter-condition-input')} ${dataCy('expression')}`)
+              .invoke('val')
+              .then((val) => {
+                expect(val).equals(filter.condition.value);
+              });
+          }
+        }
+        filter.show.map((showVal, showIndex) => {
+          cy.get(
+            `${filterSelector} ${dataCy('filter-showlist-input')} ${dataCy(
+              `show-${showIndex}`
+            )} ${dataCy('name')}`
+          )
+            .invoke('val')
+            .then((val) => {
+              expect(val).equals(showVal.name);
+            });
+
+          cy.get(
+            `${filterSelector} ${dataCy('filter-showlist-input')} ${dataCy(
+              `show-${showIndex}`
+            )} ${dataCy('type')}`
+          )
+            .invoke('val')
+            .then((val) => {
+              expect(val).equals(showVal.type);
+            });
+        });
+      });
+
+      // View live JSON output
+      cy.get(dataCy('open-live-view-button')).click({ force: true });
+
+      // Compare the live JSON code with our mock data
+      cy.get(dataCy('live-json')).should((jsonContent) => {
+        const obj = JSON.parse(jsonContent.text());
+        const filters = obj.filters;
+
+        expect(filters).to.deep.equal(MOCK_FILTERS);
+      });
+    });
+  });
+
+  describe('Importing data', () => {
+    beforeEach(() => {
+      cy.visit('/cdap/ns/default/plugincreation');
+    });
+
+    it('should populate the imported results for HTTP-batchsource.json', () => {
+      const filename = 'HTTP-batchsource.json';
+      const fileNameWithoutExtension = filename.substring(0, filename.lastIndexOf('.'));
+      const [pluginName, pluginType] = fileNameWithoutExtension.split('-');
+      cy.get(dataCy('plugin-json-import-button')).click();
+      cy.upload_plugin_json(filename, 'plugin-json-uploader');
+      cy.wait(1000);
```

Comment:
Instead of adding a wait here (though it is admittedly a short one), is it possibly to check for the existence of some element once the upload is complete?

Reply:
I tried deleting `wait` and it seems to work well!

Resolved.